svm/src/transaction_processor.rs

```diff
@@ -191,6 +191,15 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         }
     }
 
+    /// Returns the current environments depending on the given epoch
+    pub fn get_environments_for_epoch(&self, epoch: Epoch) -> ProgramRuntimeEnvironments {
+        self.program_cache
+            .read()
+            .unwrap()
+            .get_environments_for_epoch(epoch)
+            .clone()
```

Comment:
I think you can move the clone directly to ProgramCache `get_environmens_for_epoch` directly. It is only used in two places and all of them need a clone.

Reply:
Good advice. I'll update the PR.